[clkmgr] swap most clkmgr resets to lc root (#13095)
lowRISC/opentitan: hw/ip/clkmgr/doc/_index.md

![Clock Manager Block Diagram](clkmgr_block_diagram.svg)

> ### Reset Domains

Author: @matutem let me know if this makes sense to you.

Reviewer: Yes, thanks.

Since the function of the clock manager is tied closely into the power-up behavior of the device, the reset domain selection must also be purposefully done.
To ensure that default clocks are available for the [power manager to release resets and initialize memories]({{< relref "hw/ip/pwrmgr/doc/_index.md#fast-clock-domain-fsm" >}}), the clock dividers inside the clock manager directly use `por` (power-on-reset) derived resets.

> This ensures that the root clocks are freely running available after power-up and is not conditioned by any other activity in the device.

Reviewer: ```suggestion
This ensures that the root clocks are freely running after power-up, regardless of any other activity in the device.
```

Author: updated.
